frontend/lib/src/features/home_screen.dart

```diff
@@ -20,36 +22,69 @@ class HomeScreen extends StatelessWidget {
     HomeScreenController homeScreenController = Get.put(HomeScreenController());
 
     final pages = [
-      Dashboard(),
-      const MyHomePage(title: 'temporary'),
+      const SettingsScreenNavigator(),
+      const DashboardScreenNavigator(),
+      const ProfileScreenNavigator(),
     ];
 
-    return SafeArea(
-      child: Scaffold(
-        floatingActionButton: FloatingActionButton(
-          child: const Icon(
-            Icons.add,
-            color: Colors.white,
+    List<GlobalKey<NavigatorState>> navigatorKeys = [
+      settingsAppNav,
+      dashboardAppNav,
+      profileAppNav,
+    ];
+
+    Future<bool> onWillPop() {
+      if (navigatorKeys[homeScreenController.currentPage].currentState!.canPop()) {
+        navigatorKeys[homeScreenController.currentPage].currentState?.pop(
+              navigatorKeys[homeScreenController.currentPage].currentContext,
+            );
+      } else {
+        SystemChannels.platform.invokeMethod<void>('SystemNavigator.pop');
+      }
+
+      return Future(() => false);
+    }
+
+    return WillPopScope(
+      onWillPop: onWillPop,
+      child: SafeArea(
+        child: Scaffold(
+          floatingActionButton: FloatingActionButton(
+            child: const Icon(
+              Icons.add,
+              color: Colors.white,
+            ),
+            onPressed: () {},
           ),
-          onPressed: () {},
-        ),
-        bottomNavigationBar: CurvedNavigationBar(
-          backgroundColor: const Color(0xFFDAEAEE),
-          color: const Color(0xFF5D8A99),
-          items: <Widget>[
-            SvgPicture.asset('assets/svg/home.svg'),
-            SvgPicture.asset('assets/svg/profile_settings.svg'),
-          ],
-          onTap: (index) {
-            /* We can change the displayed screen here*/
-            // TODO (stephen): temporary navigation to previous homescreen
-            homeScreenController.setCurrentPage = index;
-          },
-        ),
-        /* TODO: All screens now inherit the HeaderImage template */
-        body: Obx(
-          () => HeaderImage(
-            child: pages[homeScreenController.currentPage],
+          bottomNavigationBar: CurvedNavigationBar(
+            backgroundColor: const Color(0xFFDAEAEE),
+            color: const Color(0xFF5D8A99),
+            items: <Widget>[
+              SvgPicture.asset('assets/svg/list.svg'),
+              SvgPicture.asset('assets/svg/home.svg'),
+              SvgPicture.asset('assets/svg/profile_settings.svg'),
+            ],
+            onTap: (index) {
+              homeScreenController.setCurrentPage = index;
+
+              if (index == 0) {
```

Comment:
the implementation produces weird effects

ex:
goto first tab
click 1 option
goto second tab
goto first tab

the animation is weird

expected:
goto first tab
select 1 option
goto other tabs NOT first tab, reset first tab
goto first tab no weird animation

Reply:
I thinks that's just the default animation when widget is popped.

Reply:
@egiegarciano14 yes, but the logic is what im pointing out, it should reset when its not in focus,

currently its only resetting when coming into the page